redfish/drive.go

```diff
@@ -299,6 +299,10 @@ type Drive struct {
 	// WriteCacheEnabled shall indicate whether the drive
 	// write cache is enabled.
 	WriteCacheEnabled bool
+	// Actions are all the listed actions under the drive
+	Actions DriveActions
```

Comment:
Remove this, actions shouldn't be exposed directly.

Reply:
With this removed, how should I surface the `Oem` field of the actions to the user? I could expose `ActionsOem` as a `json.RawMessage` off the `Drive`, perhaps? I'll need that `Oem` field under actions from the drive somehow exposed so the `oem/smc` package can consume it.

Reply:
It shouldn't be surfaced to the end user. Just in the oem object, there should be functions on the OEM Drive to perform the available actions. So something like:

```go
var drive redfish.Drive // Assigned elsewhere

smcDrive, err := smc.FromDrive()
smcDrive.Indicate()
```

Reply:
Understood; let the OEM Drive model perform the action. The confusion I'm having here is that the URL that's necessary is buried in the Actions, under another `Oem` field as seen in the test [here](https://github.com/stmcginnis/gofish/pull/366/files#diff-d2586b9d6c105ec199b20cfd0957883d3d6be91e06be8171843823a21f21914eR38-R45).

The `oem/smc` package needs a way to get the drive -> actions -> oem data. That's why I exposed the `Actions` struct here, or suggest exposing `ActionsOem` hanging off the `redfish.Drive`. I'm not sure what approach best fits the gofish conventions.

Reply:
Yeah, sorry, I didn't have a good example to easily show what I was thinking. I've done a couple SMC updates, so now I do have good examples.

Here's an example of accessing the properties of an OEM object: https://github.com/stmcginnis/gofish/blob/main/oem/smc/chassis.go#L27-L44

And its usage: https://github.com/stmcginnis/gofish/blob/main/oem/smc/chassis_test.go#L105

I don't have a simple example of using an action target, but it would be similar to having the function on the OEM struct like here: https://github.com/stmcginnis/gofish/blob/main/oem/smc/nodemanager.go#L100

And pulling out the action target like you would a property or link similar to the first example or here: https://github.com/stmcginnis/gofish/blob/main/oem/smc/computersystem.go#L51

Hopefully that's a little clearer. There are other examples in the standard redfish objects that show some similar approaches. Basically, I'd like to keep these OEM objects so they have a similar feel and behavior to the regular ones. I'm seeing now that some of the earlier OEM objects added did not do that, so sorry if you were using those an example. I should probably fix that. ;)

Reply:
Thanks! I wasn't finding a single answer, but a few ways to do it as you said.

In the `Chassis` one, it seems that the `RawData` is exported in the `redfish.Chassis`, which isn't the case in the `redfish.Drive`. If I change that to be exported that will make this follow that path much easier. I'll push what I have now so you can see the state it's in, and I'll clean it up a bit for the next pass.

Reply:
Perfect, thanks! I don't love having RawData exposed, but it seems like a decent trade-off to make these work easier.

Reply:
Pushed this up. I previously added the helper functions for getting the temperature etc, but I could remove those getters and simply export the values if you think that's more appropriate here.

Reply:
Just saw this comment before posting. Yes, I think that would be great.